tests/ssl-opt.sh

```diff
@@ -5489,14 +5489,14 @@ run_test    "Per-version suites: TLS 1.2" \
 
 requires_gnutls
 run_test    "ClientHello without extensions, SHA-1 allowed" \
```

Comment:
There's something wrong with these two test cases with respect to SHA-1. I think both their introduction in 5d2511c4d48eb197697466d1bd6b776cf09b0e7c and the change of certificates in 574ac577b0ec6d0b4db7268c07b03269399b4a7d don't quite do what they say in the description. Since there's no client authentication, it doesn't matter whether the server allows certificates signed by SHA-1. If it did, then the case that allows SHA-1 should not use a certificate signed with SHA-1.

This should be a single test case, using default certificates, without any mention of SHA-1 in the description, like it was before 5d2511c4d48eb197697466d1bd6b776cf09b0e7c.

Reply:
Yes, I found that weird as well, and wished the purpose and rationale for the tests was better documented, but went for the minimal change that fixed my problem.

Thanks for having a look at the history, I think it shows that the current situation was unintentional and a result of incorrectly including that test in mass-changes related to SHA-1.

I agree that it would be better to just revert to the previous state, with only one test, using the default certificates, and no mention of SHA-1. I'll amend my PR to do that.